src/terminal-screen.c

```diff
@@ -1949,7 +1949,8 @@ terminal_screen_child_exited (VteTerminal *terminal, int status)
 	switch (action)
 	{
 	case TERMINAL_EXIT_CLOSE:
-		g_signal_emit (screen, signals[CLOSE_SCREEN], 0);
+		if ((status != 9) && (priv->override_command != NULL))
```

Comment:
Replace and with or, otherwise the terminal won't be closed if the user typed `exit`.
```
if ((status != 9) || (priv->override_command != NULL))
```

Reply:
Confirmed, exit doesn't close the terminal and suggestion from Robert fixes the issue.

Reply:
yes, I forgot it, done